### PR TITLE
Made task2

### PR DESCRIPTION
**Pull request by nikitanut**
Review please.

### CODE REVIEW

**Comment by chesdenis** [comment]
@CunningPie  or @AnnaMiky can you please review/approve?

**Comment by AnnaMiky** [comment]
Sorry fo waiting)) 
It it really cool!

I think Console.SetWindowSize(81, 40); sould be in another place, because , for example, if it isn't a console app, it isn't good.
And the same with "string message" in GameLogic.

**Comment by AnnaMiky** [comment]
Sorry for waiting))
It is really cool!

I think that Console.SetWindowSize(81, 40); sould be in another place, because , for example, if it isn't a console app, it isn't good.
And the same with "string message" in GameLogic.

**Comment by nikitanut** [comment]
@AnnaMiky , thanks for the review! 
But it is said that you closed my request and didn't approve it. I'm not sure, but it looks like that my branch can't be merged, because it's closed. 
Is it possible to approve it now? Or should I reopen the request?

**Comment by AnnaMiky** [comment]
@nikitanut , Sorry.
Is it ok?

File: Task2 - Delegates/Nikita_Nuttunen/Delegates/Drawing.Core/GameLogic.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                        processUserActions.SetBoardSize(board);
+                        showMessageToUser.DrawBoard(board);
+                        showMessageToUser.ShowMessage(message);
+                        break;
+                    default:
+                        showMessageToUser.ShowMessage("Wrong Input! Please try again");
+                        break;
+                }
+                showMessageToUser.ClearInput();
+                draw?.Invoke(board);
```
Comment (AnnaMiky) [inline Task2 - Delegates/Nikita_Nuttunen/Delegates/Drawing.Core/GameLogic.cs]: Could you tell me, what is it? 
For what do you use the question mark?
Could you tell me about this way of writing?

Comment (nikitanut) [reply Task2 - Delegates/Nikita_Nuttunen/Delegates/Drawing.Core/GameLogic.cs]: At first I typed:
if (draw != null)
  {
      draw(board);
   }

Visual Studio advised me to change this code to "draw?.Invoke(board);". It's the same.
The question mark means that the variable is nullable. If you don't put it, you'll get NullReferenceException if "draw" is null.

Comment (AnnaMiky) [reply Task2 - Delegates/Nikita_Nuttunen/Delegates/Drawing.Core/GameLogic.cs]: Thank you!!

**Pull request merged**